### PR TITLE
Use random ports when port forwarding

### PR DESCRIPTION
**Pull request by kevinearls**
Signed-off-by: Kevin Earls <kearls@redhat.com>

### CODE REVIEW

**Comment by kevinearls** [comment]
@jpkrohling @pavolloffay @jpkrohling  Please review.  I'm hoping this will reduce test flakiness as we have had a lot of port forwarding errors.

File: test/e2e/utils.go [hunk 1]
```diff
@@ -213,6 +215,15 @@ func undeployJaegerInstance(jaeger *v1.Jaeger) {
 	require.NoError(t, err)
 }
 
+func randomPortNumber() string {
```
Comment (pavolloffay) [inline test/e2e/utils.go]: Random port implementations usually return a random port which is free. There is maybe a syscall which you could use to get the random port.

Note that k8s port-forward provides random port feature which I tried to use in https://github.com/jaegertracing/jaeger-operator/pull/288. To use it we would need to update k8s API to 1.15.

Comment (kevinearls) [reply test/e2e/utils.go]: @pavolloffay  I've updated this so it should now return a random port which is free.  Can you review and merge if it looks ok?

Comment (pavolloffay) [reply test/e2e/utils.go]: Could you please move this function to `portForward.go` and leave a TODO comment saying use port-forward from k8s once we upgrade k8s API to 1.15.0+

File: test/e2e/all_in_one_test.go [hunk 2]
```diff
@@ -160,11 +160,13 @@ func (suite *AllInOneTestSuite)  TestAllInOneWithUIConfig()  {
...
 
-	portForward, closeChan := CreatePortForward(namespace, "all-in-one-with-ui-config", "jaegertracing/all-in-one", []string{"16686"}, fw.KubeConfig)
+	queryPort := randomPortNumber()
+	ports := []string{queryPort + ":16686"}
+	portForward, closeChan := CreatePortForward(namespace, "all-in-one-with-ui-config", "jaegertracing/all-in-one", ports, fw.KubeConfig)
 	defer portForward.Close()
 	defer close(closeChan)
 
-	url := fmt.Sprintf("http://localhost:16686/%s/search", basePath)
+	url := fmt.Sprintf("http://localhost:" + queryPort + "/%s/search", basePath)
```
Comment (pavolloffay) [inline test/e2e/all_in_one_test.go]: there is sprintf this change be changed to

```
fmt.Sprintf("http://localhost:%d"/%s/search", queryPort, basePath)
```

File: test/e2e/utils.go [hunk 3]
```diff
@@ -1,8 +1,8 @@
 package e2e
 
 import (
-	"fmt"
 	goctx "context"
+	"fmt"
```
Comment (pavolloffay) [inline test/e2e/utils.go]: @jpkrohling test package is not being lintedm `fmt` should be before `goctx`

**Pull request merged**